### PR TITLE
stack.py Add Stack Data Structure in Python

### LINKED ISSUE #37: [DataStructure] Stack [Python]

### PR DESCRIPTION
**Pull request by shreyasbapat**
This adds Stack data structure in Python Language.
    Includes basic functionality of stack (push, pop, top, display)

Closes #37

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

Fixes #37 

**By submitting this pull request I confirm I've read and complied with the
below declarations.**

- [x] I have read the [Contribution guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [commit guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#write-good-commit-messages) for this project.
- [x] My code follows the [standard code structure](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#samples-code).
- [x] This pull request has a descriptive title. For example, `{Tag}: Add {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.
- [x] This pull request shall only be reviewed and merged once all the checks passes. No maintainer or supporter shall be obliged to review it before this condition is met.
- [x] I have mentioned the issue number correctly (with hyperlink) in this pull request description.

After you submit your pull request, **DO NOT** click the 'Update Branch' button.

### CODE REVIEW

File: stack/python/stack.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+class Stack:
+    def __init__(self):
+        self.stack = []
+
+    def push(self, dataval):
+        if dataval not in self.stack:
```
Comment (d-Rickyy-b) [inline stack/python/stack.py]: Does it really matter if the dataval is already on the stack?

Comment (shreyasbapat) [reply stack/python/stack.py]: Well, there is no guideline on what matters and what not. I will remove the line.

Comment (shreyasbapat) [reply stack/python/stack.py]: Resolved

**Comment by shreyasbapat** [comment]
@sangamcse Check again!

**Comment by sangamcse** [comment]
@sangamcse rebase

**Comment by sangamcse** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by sangamcse** [comment]
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

**Comment by sangamcse** [comment]
Changes LGTM. Please include code comments and also add to the README table in the same commit. Use `git commit --amend` for that.
Just in case, http://ohshitgit.com/

**Comment by shreyasbapat** [comment]
Done! Oh! Thanks for the resource but I knew howto!

File: stack/python/stack.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
+class Stack:
+    """
+    Class for operations related to stack.
+    """
```
Comment (sangamcse) [inline stack/python/stack.py]: The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `all.autopep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmplnn5ba1l/stack/python/stack.py
+++ b/tmp/tmplnn5ba1l/stack/python/stack.py
@@ -2,6 +2,7 @@
     """
     Class for operations related to stack.
     """
+
     def __init__(self):
         """
         Constructor.
```

File: stack/python/stack.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    def pop(self):
+        """
+        Method for removing the last element from the stack.
+
+        Returns
+        -------
+        element that is popped.
+        """
+        if len(self.stack) <= 0:
+            return ("No element in the Stack")    # Raise an error is if stack is empty.
```
Comment (sangamcse) [inline stack/python/stack.py]: The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `all.autopep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmplnn5ba1l/stack/python/stack.py
+++ b/tmp/tmplnn5ba1l/stack/python/stack.py
@@ -25,7 +25,8 @@
         element that is popped.
         """
         if len(self.stack) <= 0:
-            return ("No element in the Stack")    # Raise an error is if stack is empty.
+            # Raise an error is if stack is empty.
+            return ("No element in the Stack")
         else:
             return self.stack.pop()
 
```

File: stack/python/stack.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        else:
+            return self.stack.pop()
+
+    def top(self):
+        return self.stack[len(self.stack)-1]    # Show the last element of the list.
```
Comment (sangamcse) [inline stack/python/stack.py]: The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `all.autopep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmplnn5ba1l/stack/python/stack.py
+++ b/tmp/tmplnn5ba1l/stack/python/stack.py
@@ -30,7 +30,8 @@
             return self.stack.pop()
 
     def top(self):
-        return self.stack[len(self.stack)-1]    # Show the last element of the list.
+        # Show the last element of the list.
+        return self.stack[len(self.stack)-1]
 
     def display(self):
         print(*self.stack)
```

File: stack/python/stack.py (hunk 3 above)
Comment (sangamcse) [inline stack/python/stack.py]: E501 line too long (88 > 80 characters)

*Origin: PycodestyleBear (E501), Section: `all.autopep8`.*

File: stack/python/stack.py (hunk 4 above)
Comment (sangamcse) [inline stack/python/stack.py]: E501 line too long (84 > 80 characters)

*Origin: PycodestyleBear (E501), Section: `all.autopep8`.*

**Comment by shreyasbapat** [comment]
You want PEP8 also! Wait

File: stack/python/stack.py (hunk 4 above)
Comment (shreyasbapat) [reply stack/python/stack.py]: E501 is ignored in some core python libraries too! Do you want me to fix it? Even autopep8 ignores that xD

File: stack/python/stack.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
+class Stack:
+    """
+    Class for operations related to stack.
+    """
+
+    def __init__(self):
+
+        self.stack = [] # Initialise empty stack
+
```
Comment (sangamcse) [inline stack/python/stack.py]: The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `all.autopep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmple4xxr40/stack/python/stack.py
+++ b/tmp/tmple4xxr40/stack/python/stack.py
@@ -5,7 +5,7 @@
 
     def __init__(self):
 
-        self.stack = [] # Initialise empty stack
+        self.stack = []  # Initialise empty stack
 
     def push(self, dataval):
         """
```

Comment (sangamcse) [inline stack/python/stack.py]: E261 at least two spaces before inline comment

*Origin: PycodestyleBear (E261), Section: `all.autopep8`.*

**Comment by sangamcse** [comment]
Comment on 1351f179cd71d1635c284e6b3bfadaf59c067fb4, file stack/Python/stack.py, line 8.

The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `all.autopep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpe3wqxese/stack/Python/stack.py
+++ b/tmp/tmpe3wqxese/stack/Python/stack.py
@@ -5,7 +5,7 @@
 
     def __init__(self):
 
-        self.stack = [] # Initialise empty stack
+        self.stack = []  # Initialise empty stack
 
     def push(self, dataval):
         """
```

**Comment by sangamcse** [comment]
Comment on 1351f179cd71d1635c284e6b3bfadaf59c067fb4, file stack/Python/stack.py, line 8.

E261 at least two spaces before inline comment

*Origin: PycodestyleBear (E261), Section: `all.autopep8`.*

**Comment by sangamcse** [comment]
Comment on d6c70cd27162fe109e3d07efa67bf1dfa231ae01, file stack/Python/stack.py, line 8.

The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `all.autopep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpt__y8x3_/stack/Python/stack.py
+++ b/tmp/tmpt__y8x3_/stack/Python/stack.py
@@ -5,7 +5,7 @@
 
     def __init__(self):
 
-        self.stack = [] # Initialise empty stack
+        self.stack = []  # Initialise empty stack
 
     def push(self, dataval):
         """
```

**Comment by sangamcse** [comment]
Comment on d6c70cd27162fe109e3d07efa67bf1dfa231ae01, file stack/Python/stack.py, line 8.

E261 at least two spaces before inline comment

*Origin: PycodestyleBear (E261), Section: `all.autopep8`.*

**Comment by sangamcse** [comment]
@sangamcse rebase

**Comment by sangamcse** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by sangamcse** [comment]
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

**Pull request merged**